infra/discovery/src/listener/listener.js

```diff
@@ -269,12 +127,63 @@ const errorCounter = new bundle.promClient.Counter({
 })
 
 /**
- * Creates runtime context and starts the live tracking engine.
+ * Creates runtime context and starts the tracking loop
  * @return {Promise<void>}
  */
 async function main() {
   const context = await new Context().init(config, errorCounter)
-  liveTracking(context)
+  let lastProcessedBlock = await getLastBlock(context.config)
+  logger.info(`Resuming processing from ${lastProcessedBlock}`)
+  const checkIntervalSeconds = 5
+  let start
+
+  const check = async () => {
+    await withRetrys(async () => {
+      start = new Date()
+      const currentBlock = await context.web3.eth.getBlockNumber()
+      if (currentBlock === lastProcessedBlock) {
+        logger.debug('No new blocks to process')
+        return scheduleNextCheck()
+      }
+      blockGauge.set(currentBlock)
+
+      contractsContext.marketplace.eventCache.updateBlock(currentBlock)
+      contractsContext.identityEvents.eventCache.updateBlock(currentBlock)
+
+      const fromBlock = lastProcessedBlock + 1
+      // Respect the trailBlocks option
+      const toBlock = Math.max(currentBlock - context.config.trailBlocks, 0)
+      logger.debug(`Querying events from ${fromBlock} to ${toBlock}`)
+
+      // Retrieve all events for the relevant contracts
+      const eventArrays = await Promise.all([
+        contractsContext.marketplace.eventCache.allEvents(),
+        contractsContext.identityEvents.eventCache.allEvents()
+      ])
+
+      // Flatten array of arrays filtering out anything undefined
+      const events = [].concat(...eventArrays.filter(x => x))
+      events
+        // Filter to only new events
+        .filter(event => event.blockNumber >= lastProcessedBlock)
+        // Process each new event
+        .map(newEvent => handleEvent(newEvent, context))
+
+      // Record state of processing
```

Comment:
@nick would you mind checking out this as a way of getting new events?

Reply:
Looks good to me. I guess there is similar code inside the eventCache for checking toBlock, fromBlock etc.. so I'm not sure we need to duplicate it out here? Not sure though

Reply:
Yea it is similar but I don't think I can leverage it to just get new events like this, perhaps you know a way though.